### ISSUE TITLE
[Bug][Auto-scheduler] generated cuda code

### ISSUE THREAD

**Issue by Haoran-Solar** [apache/tvm#7732]
To be simple, the cuda code that generated by auto-scheduler from file
```
/tutorials/auto_scheduler/tune_conv2d_layer_cuda.py
```
I call it as following:
```c++
dim3 dimBlock(128);
dim3 dimGrid(16);

default_function_kernel0<<<dimGrid, dimBlock>>>(data, kernel, compute, bias);

cudaDeviceSynchronize();
```
Then bug occurred :
![image](https://user-images.githubusercontent.com/56677718/112267731-daec8b00-8cb0-11eb-892a-351cb41349e2.png)
kernel ruined cuda api? or something
If I do not run the kernel, nothing wrong.
![image](https://user-images.githubusercontent.com/56677718/112268455-f5733400-8cb1-11eb-8b3d-2b0d281b5257.png)

I cannot figure out what happened, can you help me?
P4 and T4, I used, with cuda-10.2

**Comment by comaniac** [comment]
How many trials did you try and how did you apply the schedule from the log file? It's possible for auto-scheduler to generate invalid kernels during the tuning, but they shouldn't be picked ultimately when compiling the model.

**Comment by Haoran-Solar** [comment]
Thanks for reply. I just started to try auto-scheduler, I change nothing in file tune_conv2d, just want to have a test.
Using prams from last layer in ResNet-50, just the tutorials did. 
And I solve the problem. To run the generated code, I have to set the number of each dim less than 64 in a threadblock.
Which mean:
```c++
dim3 dimBlock(number);
*number<64*
*64, 32,16 ....*
```
Is that some special limit for auto-scheduler code? Cause some kernels that I wrote do not such limit.
